src/sirplugins.c

```diff
@@ -255,11 +255,11 @@ bool _sir_plugin_unload(sir_plugin* plugin) {
                      " cleanup!", plugin->path, plugin->handle);
 
 # if !defined(__WIN__)
-    if (0 != dlclose(plugin->handle)) {
+    if (0 != dlclose(plugin->handle)) { // GCOVR_EXCL_START
```

Comment:
when you add these, do you brace yourself for the inevitable comment?

Reply:
💀 but no way to force the test really for this path :(

Reply:
Well, it would probably be less ghetto to add some macro like `SIR_PURPOSEFUL_ERRORS` and put code inside that. For example, here, set the handle to null or unload the module before the test. It's cleaner (and also tests the code path) than putting these comments everywhere that GCOVR just happens to never hit.

Reply:
That's not what I did though.  There are over a hundred\*\*\* places that gcovr doesn't hit, but I didn't put exclusions them, only on the specific places where it's apparent that it is not possible to create a failure in any simple way at runtime.  

\*\*\* 

```
src/sirerrors.c                              105     104    99%   237
src/sirmutex.c                                31      30    96%   50
example/example.c                             49      46    93%   66,92,176
src/sirqueue.c                                65      62    95%   39,65,93
src/sirtextstyle.c                            94      91    96%   73,119,173
include/sir.hh                               257     254    98%   712,728,826
include/sir/helpers.h                         82      78    95%   123-124,382,418
tests/tests++.cc                             211     204    96%   116-122
tests/tests_shared.c                         183     175    95%   165,183,211-212,246,258,278,312
src/sirhelpers.c                             191     180    94%   48-49,51,59,88,92,294,310,316,354-355
src/sirthreadpool.c                          103      92    89%   47,55-57,64-66,73-75,138
src/sirfilesystem.c                          127     115    90%   46,136,152,180,247,264,267-269,430,436,625
src/sirfilecache.c                           329     315    95%   42,83,146-147,187,199,242,260,307,329,339-340,631,644
src/sirplugins.c                             198     178    89%   80,209,216,245,283,300,309,318,342,349,371,381,389,401,408,431,438,451,460,500
tests/tests.c                               1214    1190    98%   1452,1580,1589-1590,1708-1709,1718,1781,1786-1787,1801-1802,2219,2250-2252,2270,2357,2461,2467,2513-2514,2526-2527
src/sirinternal.c                            593     554    93%   171-172,176-177,199-200,236-237,245-246,286,434,440,615,687,864,870,912,919-920,924-925,956,963-964,968-969,1057,1064,1107,1111,1118,1152,1236,1241,1326,1368,1475,1481
------------------------------------------------------------------------------
```